src/Microsoft.AspNet.Server.Kestrel/Http/Frame.cs

```diff
@@ -775,6 +775,11 @@ protected bool TakeStartLine(SocketInput input)
                     // URI was encoded, unescape and then parse as utf8
                     pathEnd = UrlPathDecoder.Unescape(pathBegin, pathEnd);
                     requestUrlPath = pathBegin.GetUtf8String(pathEnd);
+
+                    if (PathNormalizer.NeedsNormalization(requestUrlPath))
```

Comment:
needDecode and NeedsNormalization are now out of sync because dots don't trigger needDecode. Add a functional test because I don't think this code is actually executing right now.